internal/controller/issues.go

```diff
@@ -0,0 +1,301 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	issuesv1alpha1 "github.com/matanamar10/github-issue-operator-hhome-assignment/api/v1alpha1"
+	"github.com/matanamar10/github-issue-operator-hhome-assignment/internal/finalizer"
+	"github.com/matanamar10/github-issue-operator-hhome-assignment/internal/git"
+	"go.uber.org/zap"
```

Comment:
You need to break up this file to smaller files each having a specific function. Don't be shy moving some of the logic to the main `controller.go` file as well since you kept it so bare and just moved all the logic to a single file elsewhere which doesn't really achieve anything